src/SDSS.jl

```diff
@@ -130,5 +135,454 @@ function load_stamp_catalog(cat_dir, stamp_id, blob; match_blob=false)
+
+ This is based on the function setMaskedPixels in astrometry.net:
+ https://github.com/dstndstn/astrometry.net/
+""" ->
+function mask_image!(mask_img, field_dir, run_num, camcol_num, frame_num, band;
+                     python_indexing = false,
+                     mask_planes = Set({"S_MASK_INTERP", "S_MASK_SATUR", "S_MASK_CR", "S_MASK_GHOST"}))
+    # The default mask planes are those used by Dustin's astrometry.net code.    
+    # See the comments in sdss/dr8.py for fpM.setMaskedPixels
+    # and the function sdss/common.py:fpM.setMaskedPixels
+    #
+    # interp = pixel was bad and interpolated over
+    # satur = saturated
+    # cr = cosmic ray
+    # ghost = artifact from the electronics.
+
+    # http://data.sdss3.org/datamodel/files/PHOTO_REDUX/RERUN/RUN/objcs/CAMCOL/fpM.html
+    band_letter = bands[band]
+    fpm_filename = "$field_dir/fpM-$run_num-$band_letter$camcol_num-$frame_num.fit"
+    fpm_fits = FITS(fpm_filename)
+
+    # The last header contains the mask.
+    fpm_mask = fpm_fits[12]
+    fpm_hdu_indices = read(fpm_mask, "value")
+
+    # Only these rows contain masks.
+    masktype_rows = find(read(fpm_mask, "defName") .== "S_MASKTYPE")
+
+    # Apparently attributeName lists the meanings of the HDUs in order.
+    mask_types = read(fpm_mask, "attributeName")
+    plane_rows = findin(mask_types[masktype_rows], mask_planes)
+
+    # Make sure each mask is present.  Is this check appropriate for all mask files?
+    @assert length(plane_rows) == length(mask_planes)
+
+    for fpm_i in plane_rows
+        # You want the HDU in 2 + fpm_mask.value[i] for i in keep_rows (in a 1-indexed language).
+        mask_index = 2 + fpm_hdu_indices[fpm_i]
+        cmin = read(fpm_fits[mask_index], "cmin")
+        cmax = read(fpm_fits[mask_index], "cmax")
+        rmin = read(fpm_fits[mask_index], "rmin")
+        rmax = read(fpm_fits[mask_index], "rmax")
+        row0 = read(fpm_fits[mask_index], "row0")
+        col0 = read(fpm_fits[mask_index], "col0")
+
+        @assert all(col0 .== 0)
+        @assert all(row0 .== 0)
+        @assert length(rmin) == length(cmin) == length(rmax) == length(cmax)
+
+        for block in 1:length(rmin)
+            # The ranges are for a 0-indexed language.
+            @assert cmax[block] + 1 <= size(mask_img)[1]
+            @assert cmin[block] + 1 >= 1
+            @assert rmax[block] + 1 <= size(mask_img)[2]
+            @assert rmin[block] + 1 >= 1
+
+            # Some notes:
+            # See astrometry.net//sdss/common.py:SetMaskedPixels, which I currently assume is correct.
+            # - In contrast with  julia, the numpy matrix index range [3:5, 3:5] contains four
+            #   pixels, not six.  However, if the numpy is correct, then fpM files contain
+            #   many bad rows that don't get masked at all since cmin == cmax
+            #   or rmin == rmax.  For this reason, I think the python might be erroneous.
```

Comment:
It was a soft confirmation: he agrees that it might be a bug.  I'm sure enough to make this way the default, but let's leave this note for now.